src/us/kbase/workspace/test/database/UtilTest.java

```diff
@@ -0,0 +1,222 @@
+                TestCommon.assertExceptionCorrect(got, new NullPointerException(OK_STRING));
+            }
+        }
+    }
+
+    @Test
+    public void passCheckNoNullsOrEmpties() throws Exception {
+        Util.checkNoNullsOrEmpties(nonEmptyStrings, TYPE_NAME);
+    }
+
+    @Test
+    public void failCheckNoNullsOrEmpties() throws Exception {
+        final List<List<String>> testList = new ArrayList<>();
+        testList.add(emptyStrings);
+        testList.add(emptyStringsWithNull);
+        testList.add(nonEmptyStringsWithNull);
+
+        for (List<String> testListItem : testList) {
+            try {
+                Util.checkNoNullsOrEmpties(testListItem, TYPE_NAME);
+                fail(EXP_EXC);
+            } catch (Exception got) {
+                TestCommon.assertExceptionCorrect(got, new IllegalArgumentException(
+                        "Null or whitespace only string in collection " + TYPE_NAME));
+            }
+        }
+    }
+
+    @Test
+    public void passCheckString() throws Exception {
+        final String INCORRECT_CHECKSTRING = "incorrect checkString";
+        final Map<String, String> trimmedNonEmptyStrings = new HashMap<>();
+        for (String nonEmpty : nonEmptyStrings) {
+            trimmedNonEmptyStrings.put(nonEmpty, nonEmpty.trim());
+        }
+
+        for (Map.Entry<String, String> mapElement : trimmedNonEmptyStrings.entrySet()) {
+            assertThat(
+                    INCORRECT_CHECKSTRING,
+                    Util.checkString(mapElement.getKey(), TYPE_NAME),
+                    is(mapElement.getValue()));
+        }
+
+        final int max = 100;
+        for (Map.Entry<String, String> mapElement : trimmedNonEmptyStrings.entrySet()) {
+            assertThat(
+                    INCORRECT_CHECKSTRING,
+                    Util.checkString(mapElement.getKey(), TYPE_NAME, max),
+                    is(mapElement.getValue()));
+            // max is only taken into account when larger than zero
+            assertThat(
+                    INCORRECT_CHECKSTRING,
+                    Util.checkString(mapElement.getKey(), TYPE_NAME, 0),
+                    is(mapElement.getValue()));
+            assertThat(
+                    INCORRECT_CHECKSTRING,
+                    Util.checkString(mapElement.getKey(), TYPE_NAME, -20),
+                    is(mapElement.getValue()));
+        }
+    }
+
+    @Test
+    public void failCheckStringNullWhitespace() throws Exception {
+        for (String emptyNullString : emptyStringsWithNull) {
+            try {
+                Util.checkString(emptyNullString, TYPE_NAME);
+                fail(EXP_EXC);
+            } catch (Exception got) {
+                TestCommon.assertExceptionCorrect(got, new IllegalArgumentException(
+                        TYPE_NAME + " cannot be null or whitespace only"));
+            }
+        }
+    }
+
+    @Test
+    public void failCheckStringTooLong() throws Exception {
+        final int max = 1;
```

Comment:
I would make this 1 smaller than the largest string in the set

Reply:
Fixed

Reply:
It's still 1 though?

Reply:
All these tests should fail because the string is too long; the shortest string in the test set is two code points long.

Reply:
:+1: